sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_test.cpp

```diff
@@ -572,7 +572,33 @@ TEST_F(KeyVaultCertificateClientTest, BackupRestoreCertificate)
     auto result = response.PollUntilDone(m_defaultWait);
     EXPECT_EQ(result.Value.Name(), certificateName);
     client.PurgeDeletedCertificate(certificateName);
-    std::this_thread::sleep_for(m_defaultWait);
+    TestSleep(m_defaultWait);
+  }
+  {
+    int retries = 15;
+    // before we restore we need to ensure the certificate is purged.
+    // since we have no visibility into the purge status we will just wait and check for a few
+    // seconds
+    while (retries > 0)
+    {
+      try
+      {
+        client.GetDeletedCertificate(certificateName);
+      }
+      catch (Azure::Core::RequestFailedException const&)
+      {
+        std::cout << std::endl << "- Certificate is gone";
+        break;
+      }
+      TestSleep(m_defaultWait);
+      retries--;
+    }
+
+    if (retries == 0)
+    {
+      std::cout << std::endl << "retries reached 0";
+      EXPECT_TRUE(false);
```

Comment:
FYI: Another mechanism to add output messages for tests that fail, instead of `cout`, is use the gtest feature.

This way, you only output that error message if the test assertion fails.

You can also use: `GTEST_LOG_INFO`. We have a few places in test that do that already.

In cases like this, where you want to fail a test if it reaches this point, rather than using `EXPECT_TRUE(false)`, you can see `GTEST_FAIL` with an output message appended to it

Here are a couple example usages of that:
https://github.com/Azure/azure-sdk-for-cpp/blob/d609c004123f81c9a2a2b3d00095f0cd2395113d/sdk/eventhubs/azure-messaging-eventhubs/test/ut/eventhubs_admin_client_test.cpp#L68-L69
https://github.com/Azure/azure-sdk-for-cpp/blob/d609c004123f81c9a2a2b3d00095f0cd2395113d/sdk/identity/azure-identity/test/ut/azure_pipelines_credential_test.cpp#L573